shell/platform/darwin/common/framework/Source/FlutterStandardCodec.mm

```diff
@@ -45,7 +45,7 @@ - (NSData*)encode:(id)message {
 }
 
 - (id)decode:(NSData*)message {
-  if (message == nil)
+  if (message == nil || [message length] == 0)
     return nil;
```

Comment:
Have you checked all of our other codecs to see if they have the same logic?

Reply:
I believe the answer to that one is 'no' though my zsh history indicates that grepped for all of them then promptly ignored all the others, which to be honest, makes this even more criminal. Fixed and tests added.

FWIW I've noticed that at least our Java implementations for Android have a similar issue. Will put together a separate patch for those.

Reply:
Worth noting that FlutterStringCodec already handles zero-length messages but interprets them as a zero length string rather than nil; fair enough given the codec in question.